src/pool/srv_pool.c

```diff
@@ -2935,12 +2983,20 @@ ds_pool_query_info_handler(crt_rpc_t *rpc)
 	tgt_state = target->ta_comp.co_status;
 	out->pqio_state = enum_pool_comp_state_to_tgt_state(tgt_state);
 
-	/**
-	 * TODO (DAOS-3625): Send pool tgt query RPC (server->server) to
-	 * return pool target space info (including fragmentation).
-	 */
-
 	ABT_rwlock_unlock(svc->ps_pool->sp_lock);
+
+	if (tgt_state == PO_COMP_ST_UPIN) {
+		rc = pool_query_tgt_space(rpc->cr_ctx, svc, in->pqii_op.pi_hdl,
+					  in->pqii_rank, in->pqii_tgt,
+					  &out->pqio_space);
+		if (rc)
+			D_ERROR(DF_UUID": Failed to query rank:%u, tgt:%d, "
+				""DF_RC"\n", DP_UUID(in->pqii_op.pi_uuid),
+				in->pqii_rank, in->pqii_tgt, DP_RC(rc));
+	} else {
+		memset(&out->pqio_space, 0, sizeof(out->pqio_space));
+	}
+
 	pool_svc_put_leader(svc);
```

Comment:
I think the code needs to add a call to ds_rsvc_set_hint() call to populate the pool query output po_hint field. This will help clients in their logic that keeps track of who the leader is.
ds_rsvc_set_hint(&svc->ps_rsvc, &out->pqio_op.po_hint);

Reply:
ok